### PR TITLE
Add basic terraform structure and an AWS EKS cluster tf module

### PR DESCRIPTION
**Pull request by MOZGIII**
This PR adds terraform modules, that will be responsible for managing the AWS resources.

In this PR we:
- wrap the cluster creation via `eksctl` with a terraform provider
- wrap the flux initial bootstrap with a shell script that applies the kustomization via `kubectl` through a shell script terraform provider
- expose all this as a set of terraform modules, intended to be included as child modules into a private, "real" root module

### CODE REVIEW

File: terraform/aws/main.tf [hunk 1]
```diff
@@ -0,0 +1,62 @@
+provider "eksctl" {}
```
Comment (jszwedko) [inline terraform/aws/main.tf:1]: Does this top level config just exist as an example or are you using it to actually manage resources?

**Review by jszwedko — COMMENTED** [review]
This is cool!

File: terraform/aws/main.tf (hunk 1 above)
Comment (MOZGIII) [reply terraform/aws/main.tf:1]: It has to be present to be properly passed along to the child modules. It's just an inconvenience with terraform modules system, but I think those providers have to be there. I tried removing them before, but got some errors, so I decided to leave them.

I'm not a big fan of how we have to duplicate version requirements everywhere.

**Comment by MOZGIII** [comment]
Merging since we're in the very early stages and I'm finally satisfied with the approach.
Feedback is welcome!

**Pull request merged**